samples/meeting-live-coding-interview/csharp/MeetingLiveCoding/Controllers/HomeController.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MeetingLiveCoding.Models;
+using System.Collections.Concurrent;
+
+namespace MeetingLiveCoding.Controllers
+{
+    [Route("api/editorState")]
+    [ApiController]
+    public class HomeController : Controller
+    {
+        private readonly ConcurrentDictionary<string, List<MeetingDetails>> _meetingDetails;
+
+        public HomeController(
+            ConcurrentDictionary<string, List<MeetingDetails>> meetingDetails)
+        {
+            _meetingDetails = meetingDetails;
+        }
+
+        /// <summary>
+        /// Method to get all the questions set for a meeting.
+        /// </summary>
+        /// <param name="questionId"></param>
+        /// <param name="meetingId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        public async Task<IActionResult> GetMeetingDetails(string questionId, string meetingId)
+        {
+            try
+            {
+                var questions = meetingId;
+                if (questions == null) return NotFound();
+
+                var latestValue = new EditorValue();
+                var currentMeetingList = new List<MeetingDetails>();
+                var meetingList = new List<MeetingDetails>();
+                _meetingDetails.TryGetValue("meetingDetails", out currentMeetingList);
+
+                if (currentMeetingList == null)
+                {
+                    var meetingDetails = new MeetingDetails()
+                    {
+                        MeetingId = meetingId,
+                        Questions = new List<Question>
+                        {
+                           new Question()
+                           {
+                               QuestionId = "1",
+                               Value = ""
+                           },
+                           new Question()
+                           {
+                               QuestionId = "2",
+                               Value = ""
+                           },
+                           new Question()
+                           {
+                               QuestionId = "3",
+                               Value = ""
+                           }
+                        }
+                    };
+
+                    meetingList.Add(meetingDetails);
+                    _meetingDetails.AddOrUpdate("meetingDetails", meetingList, (key, newvalue) => currentMeetingList);
```

Comment:
Here, we can AddORUpdate directly as it supports key null check inside it. Please see once.